components/SearchBar.vue

```diff
@@ -2,92 +2,180 @@
+await locationsStore.fetchLocations();
 
-const languageOptions = [{
-    code: '',
-    simpleText: '----Any----',
-    displayText: '----Any----'
-}, ...localeStore.mvpLocaleDisplayOptions]
+const languageOptions = [
+    {
+        code: "",
+        simpleText: "----Any----",
+        displayText: "----Any----",
+    },
+    ...localeStore.mvpLocaleDisplayOptions,
+];
 
-const locationDropdownOptions: Ref<string[]> = ref(locationsStore.citiesDisplayOptions)
-const specialtyDropdownOptions: Ref<SpecialtyDisplayOption[]> = ref(specialtiesStore.specialtyDisplayOptions)
-const languageDropdownOptions: Ref<LocaleDisplay[]> = ref(languageOptions)
+const locationDropdownOptions: Ref<string[]> = ref(
+    locationsStore.citiesDisplayOptions
+);
+const specialtyDropdownOptions: Ref<SpecialtyDisplayOption[]> = ref(
+    specialtiesStore.specialtyDisplayOptions
+);
+const languageDropdownOptions: Ref<LocaleDisplay[]> = ref(languageOptions);
 
-const selectedSpecialty: Ref<Specialty | String> = ref('')
-const selectedLocation: Ref<string> = ref('')
-const selectedLanguage: Ref<Locale | String> = ref('')
+const selectedSpecialty: Ref<Specialty | String> = ref("");
+const selectedLocation: Ref<string> = ref("");
+const selectedLanguage: Ref<Locale | String> = ref("");
+
+function createCityTable() {
+    let lookupCityCount: { [key: string]: number } = {};
+    const cities = locationsStore.citiesDisplayOptions;
+
+    for (let i = 0; i < cities.length; i++) {
+        if (cities[i] === "----Any----") {
+            continue;
+        }
+        const city = cities[i];
+        if (lookupCityCount[city]) {
+            lookupCityCount[city] += 1;
+        } else {
+            lookupCityCount[city] = 1;
+        }
+    }
+
+    return lookupCityCount;
+}
+
+const locationLookUpForDisplay = createCityTable();
 
 watchEffect(() => {
-    locationDropdownOptions.value = locationsStore.citiesDisplayOptions
-})
+    locationDropdownOptions.value = locationsStore.citiesDisplayOptions;
+});
 
 async function search() {
-    const blankRemovedLocation = selectedLocation.value == '----Any----' ? '' : selectedLocation.value
-    const blankRemovedSpecialty = selectedSpecialty.value == '----Any----' ? undefined : selectedSpecialty.value as Specialty
-    const blankRemovedLanguage = selectedLanguage.value == '----Any----' ? undefined : selectedLanguage.value as Locale
+    const blankRemovedLocation =
+        selectedLocation.value == "----Any----" ? "" : selectedLocation.value;
+    const blankRemovedSpecialty =
+        selectedSpecialty.value == "----Any----"
+            ? undefined
+            : (selectedSpecialty.value as Specialty);
+    const blankRemovedLanguage =
+        selectedLanguage.value == "----Any----"
+            ? undefined
+            : (selectedLanguage.value as Locale);
 
-    await searchResultsStore.search(blankRemovedLocation, blankRemovedSpecialty, blankRemovedLanguage)
+    await searchResultsStore.search(
```

Comment:
This formatting all looks completely different. Let's figure this out before we merge

Reply:
I put everything back the way it was